### PR TITLE
Open sourcing of Cell-DINO and Channel-Adaptive-DINO

### PR DESCRIPTION
**Pull request by coupriec**
# Summary

This PR introduces code to reproduce main results described in the papers "Scaling Channel-Adaptive Self-Supervised Learning", published in TMLR a few months ago https://openreview.net/forum?id=pT8sgtRVAf , and to the paper "Cell-DINO: Self-Supervised Image-based Embeddings for Cell Fluorescent Microscopy" , in revision. 
The code contains a main README_CELL-DINO_AND_CHANNEL-DINO.md, that points to two separate readme corresponding to the two papers. 

We list the changes in 3 categories:

## 1- New non .py files or documentation: License, readmes, scripts to launch code to reproduce some results, and configuration files.

LICENSE_CELLDINO
LICENSE_CELLDINO_WEIGHTS
README.md
README_CELL-DINO_AND_CHANNEL-DINO.md

docs/Cell-DINO.png
docs/README_CELLDINO.md
docs/README_CHANNEL_ADAPTIVE_DINO.md
scripts/cell_dino/launcher_CHAMMI_eval.sh
scripts/cell_dino/launcher_knn_eval_on_chammi.sh
notebooks/cell_dino/inference.py

dinov2/configs/eval/cell_dino/vitl16_channel_adaptive_pretrain.yaml  
dinov2/configs/eval/cell_dino/vitl16_pretrain.yaml
dinov2/configs/train/cell_dino/vitl16_boc_hpafov.yaml  
dinov2/configs/train/cell_dino/vitl16_hpafov.yaml  
dinov2/configs/train/cell_dino/vitl16_hpaone.yaml

## 2- New .py files: data and evaluation files

These new files contain data or checkpointing helpers (dinov2/data/accumulators.py), dinov2/utils/checkpoint.py : necessary function to the linear_celldino eval), specific transforms used for pretraining (cell_augmentations.py), for evaluations (dinov2/data/cell_dino/transforms.py ), new data loaders ( in dinov2/data/datasets/cell_dino : chammi_cp.py chammi_hpa.py chammi_wtc.py hpafov.py hpaone.py)

Specific knn and linear evaluation files have been created with new options:

dinov2/eval/cell_dino/utils.py
dinov2/eval/cell_dino/knn.py
dinov2/eval/cell_dino/linear.py
dinov2/run/eval/cell_dino/knn.py
dinov2/run/eval/cell_dino/linear.py

dinov2/hub/cell_dino/backbones.py: 	introducing Celldino models

## 3- Modified files

dinov2/configs/ssl_default_config.yaml

dinov2/data/__init__.py : 	added necessary imports
dinov2/data/adapters.py : 	implemented padding in DatasetWithEnumeratedTargets
dinov2/data/datasets/__init__.py: added necessary imports
dinov2/data/datasets/decoders.py : introduced new decoders necessary to load multi-channel images
dinov2/data/datasets/extended.py : introduced options to use different decoders
dinov2/data/loaders.py : 	added necessary imports to use the new datasets

dinov2/eval/metrics.py : 	added multi-label and multi-class F1 score metrics

dinov2/models/__init__.py: 	propagate the in_chans (number of channels) and channel_dataptive parameters in build_model
dinov2/models/vision_transformer.py: implemented the channel adaptive option

dinov2/train/train.py:	picking the correct pretraining augmentation from the config option train.cell_augmentation
dinov2/utils/cluster.py : 	useful modification to test this PR.

### CODE REVIEW

**Review by patricklabatut — APPROVED** [review]
LGTM, thanks!
